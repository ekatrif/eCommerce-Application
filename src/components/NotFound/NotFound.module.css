#notfound {
  position: relative;
  height: 65vh;
}

.notfound {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
      -ms-transform: translate(-50%, -50%);
          transform: translate(-50%, -50%);
}

.notfound {
  max-width: 560px;
  width: 100%;
  padding-left: 160px;
  line-height: 1.1;
}

.notfound .notfound_404 {
  position: absolute;
  left: 0;
  top: 0;
  display: inline-block;
  width: 140px;
  height: 140px;
  background-image: url('../../assets/images/emoji.png');
  background-size: cover;
}

.notfound .notfound_404:before {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-transform: scale(2.4);
      -ms-transform: scale(2.4);
          transform: scale(2.4);
  border-radius: 50%;
  background-color: #247c5273;
  z-index: -1;
}

.notfound h1 {
  font-family: 'Nunito', sans-serif;
  font-size: 65px;
  font-weight: 700;
  margin-top: 0px;
  margin-bottom: 10px;
  color: #151723;
  text-transform: uppercase;
}

.notfound h2 {
  font-family: 'Nunito', sans-serif;
  font-size: 21px;
  font-weight: 400;
  margin: 0;
  text-transform: uppercase;
  color: #151723;
}

.notfound p {
  font-family: 'Nunito', sans-serif;
  color: #151723;
  font-weight: 400;
  margin-top: 1rem;
}

.notfound a {
  margin-top: 0.5rem;
  font-family: 'Nunito', sans-serif;
  display: inline-block;
  font-weight: 700;
  border-radius: 40px;
  text-decoration: none;
  color: #388dbc;
  padding: 15px;
  background-color: #000;
}



.notfound a:active {
  scale: 0.85;
}

@media only screen and (max-width: 767px) {
  .notfound .notfound_404 {
    width: 110px;
    height: 110px;
  }
  .notfound {
    padding-left: 15px;
    padding-right: 15px;
    padding-top: 110px;
  }
}

@media only screen and (min-width: 768px) {
  .notfound a:hover {
    color: #247C52;
  }
}