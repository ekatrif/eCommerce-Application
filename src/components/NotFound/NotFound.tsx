import React from 'react';
import { Link } from 'react-router-dom';
import classes from './NotFound.module.css';

const NotFound: React.FC = () => {
  return (
    <div id={classes.notfound}>
      <div className={classes.notfound}>
        <div className={classes.notfound_404}></div>
        <h1>404</h1>
        <h2>Oops! Page Not Be Found</h2>
        <p>
          Sorry but the page you are looking for does not exist, have been removed. name changed or is temporarily
          unavailable
        </p>
        <Link to="/">Back to homepage</Link>
      </div>
    </div>
  );
};

export default NotFound;
