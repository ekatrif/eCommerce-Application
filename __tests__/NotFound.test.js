import React from 'react';
import renderer from 'react-test-renderer';
import NotFound from '../src/components/NotFound/NotFound';
import { BrowserRouter } from 'react-router-dom'

test('NotFound renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><NotFound /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
