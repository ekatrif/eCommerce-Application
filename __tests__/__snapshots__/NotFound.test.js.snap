// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders correctly 1`] = `
<div
  id="notfound"
>
  <div
    className="notfound"
  >
    <div
      className="notfound_404"
    />
    <h1>
      404
    </h1>
    <h2>
      Oops! Page Not Be Found
    </h2>
    <p>
      Sorry but the page you are looking for does not exist, have been removed. name changed or is temporarily unavailable
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      Back to homepage
    </a>
  </div>
</div>
`;
